dbt/adapters/clickhouse/credentials.py

```diff
@@ -16,18 +16,21 @@ class ClickHouseCredentials(Credentials):
     host: str = 'localhost'
     port: Optional[int] = None
     user: Optional[str] = 'default'
+    retries: int = 1
     database: Optional[str] = None
-    database_engine: Optional[str] = None
     schema: Optional[str] = 'default'
     password: str = ''
     cluster: Optional[str] = None
+    database_engine: Optional[str] = None
+    clickhouse_cloud: bool = False
```

Comment:
cluster_mode is a little misleading because it will only work with clusters that use the Replicated database engine, and the `database_replicated_enforce_synchronous_settings` is currently experimental and undocumented.  That said I agree that changing the name makes sense, and I'll try to make the limitations clear in the documentation.